### ISSUE TITLE
Unexpected behavior with Many-to-many relations based on MySQL BigInt fields

### ISSUE THREAD

**Issue by itsharshag** [prisma/prisma#11322]
### Bug description

Queries on Implicit many-to-many relations with BigInt as the foreign keys don't produce the expected output

### How to reproduce

```
git clone https://github.com/itsharshag/prisma-playground

# Bring up mysql container for testing
docker-compose up

cd starter
yarn install
npx prisma migrate dev

npx ts-node ./insert.ts
npx ts-node ./query.ts
```

### Expected behavior

Received output
```
[
   {
      "id": "1",
      "title": "Blog",
      "code": "<h1>Hello World</h1>",
      "source": "https://google.com",
      "categories": []
   },
   {
      "id": "2",
      "title": "Blog 2",
      "code": "<h1>Hello World</h1>",
      "source": "https://google.com",
      "categories": [
         {
            "id": "1",
            "name": "Blog"
         }
      ]
   }
]
```

Expected output
```
[
   {
      "id": "1",
      "title": "Blog",
      "code": "<h1>Hello World</h1>",
      "source": "https://google.com",
      "categories": [
        {
            "id": "1",
            "name": "Blog"
        }
      ]
   },
   {
      "id": "2",
      "title": "Blog 2",
      "code": "<h1>Hello World</h1>",
      "source": "https://google.com",
      "categories": [
         {
            "id": "1",
            "name": "Blog"
         }
      ]
   }
]
```

Problem goes away when I use Int instead of BigInt for the @id columns

### Prisma information

Contained in the git repo

### Environment & setup

- OS: 
- Database: 
- Node.js version: 
OS: Windows
Database: MySQL
Node.js version: v14.18.1

### Prisma Version

```
prisma                  : 3.8.1
@prisma/client          : 3.8.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules\@prisma\engines\migration-engine-windows.exe)        
Introspection Engine    : introspection-core 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules\@prisma\engines\introspection-engine-windows.exe)      
Format Binary           : prisma-fmt 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f
Studio                  : 0.452.0
```

**Comment by pantharshit00** [comment]
Thanks for the report. I can confirm this issue, looks a comparison bug in the engines.

**Comment by zackpollard** [comment]
Just found this issue after discovering that my bug report #12206 was specifically related to bigint's. I'll leave it open for now to get eyes on it but feel free to close it as a duplicate. It has information around when this bug appeared as bigint did originally used to work correctly

**Comment by SevInf** [comment]
This is reproducible with v3.8.1, but not with the latest, v4.3.1. It appears to have been fixed in 4.0.
I am going to close this for now, feel free to reopen if you manage to reproduce it on latest Prisma version.
